internal/runners/activate/activation.go

```diff
@@ -75,19 +78,34 @@ func activationLoop(out output.Outputer, subs subshell.SubShell, targetPath stri
 	return nil
 }
 
-type activateFunc func(proj *project.Project, out output.Outputer, subs subshell.SubShell) (keepGoing bool, err error)
+type activateFunc func(proj *project.Project, out output.Outputer, config defact.DefaultConfigurer, subs subshell.SubShell, setDefault bool) (keepGoing bool, err error)
 
 // activate will activate the venv and subshell. It is meant to be run in a loop
 // with the return value indicating whether another iteration is warranted.
-func activate(proj *project.Project, out output.Outputer, subs subshell.SubShell) (bool, error) {
+func activate(proj *project.Project, out output.Outputer, cfg defact.DefaultConfigurer, subs subshell.SubShell, setDefault bool) (bool, error) {
 	projectfile.Reset()
 	venv := virtualenvironment.Get()
+
 	venv.OnDownloadArtifacts(func() { out.Notice(locale.T("downloading_artifacts")) })
 	venv.OnInstallArtifacts(func() { out.Notice(locale.T("installing_artifacts")) })
 	venv.OnUseCache(func() { out.Notice(locale.T("using_cached_env")) })
-	fail := venv.Activate()
-	if fail != nil {
-		return false, locale.WrapError(fail, "error_could_not_activate_venv", "Could not activate project. If this is a private project ensure that you are authenticated.")
+
+	logging.Debug("Setting up virtual Environment")
+	if strings.ToLower(os.Getenv(constants.DisableRuntime)) == "true" {
```

Comment:
This doesn't make any sense, you are skipping venv setup when in fact the constant only calls for skipping the runtime activation.

Reply:
Well, maybe the term `runtime activation` is well defined, but what we were and still are skipping, is the download of the artifacts and all parsing of the environment variables.  

That made sense to me, as the second part is only possible if the first one completed.  We could make the logic more complicated here if `DisableRuntime` is `"true"`), such that the environment variables are still parsed if they are available ie., if the artifact have been downloaded before. But that is basically the same behavior as `DiableRuntime=="false"` and the artifacts installed.  

I believe that this is correct `DisableRuntime==true` implies the disabling the parsing of the environment variables.  That's how it has been, BTW.  I tried to stay compatible here, not messing with existent logic...